File: msa_k8s/api-gateway/src/main/java/com/example/commerce/gateway/ApiGatewayApplication.java
package com.example.commerce.gateway;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * 커머스 외부 요청의 단일 진입점을 제공하는 API Gateway 애플리케이션이다.
 */
@SpringBootApplication
public class ApiGatewayApplication {

    /**
     * Spring 컨테이너가 애플리케이션 구성 클래스를 생성할 때 사용한다.
     */
    public ApiGatewayApplication() {
    }

    /**
     * API Gateway 애플리케이션을 시작한다.
     *
     * @param args 실행 인자
     */
    public static void main(String[] args) {
        SpringApplication.run(ApiGatewayApplication.class, args);
    }
}

File: kotlin_msa/api-gateway/src/main/kotlin/com/example/commerce/gateway/ApiGatewayApplication.kt
package com.example.commerce.gateway

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

/**
 * API Gateway 애플리케이션의 진입점입니다.
 *
 * 외부 요청을 커머스 마이크로서비스 내부 API로 라우팅합니다.
 */
@SpringBootApplication
class ApiGatewayApplication

fun main(args: Array<String>) {
    runApplication<ApiGatewayApplication>(*args)
}

